conbench/hacks.py

```diff
@@ -1,49 +1,97 @@
-def get_case(tags, include_dataset=False):
-    case = [
+from typing import Dict, List
+
+CASE_KEYS_DO_NOT_USE_FOR_NAME = ("id", "name", "suite", "source", "dataset")
+
+
+def get_case_kvpair_strings(tags: Dict[str, str], include_dataset=False) -> List[str]:
```

Comment:
I realize that `Dict[str, str]` might not be true, I think the values can be arbitrary JSON types, i.e. nesting is allowed?

Reply:
I think that is technically allowed, but I can't think of a place where we do that or use that. So maybe we can tighten that restriction?

Reply:
> So maybe we can tighten that restriction?

I do hope so! Created https://github.com/conbench/conbench/issues/940

Reply:
The only nested one I know of offhand is [array altrep materialization](https://github.com/voltrondata-labs/arrowbench/blob/ebfab3d5535a1a7ba8680be4d0394296b9b1a5ee/R/bm-array-altrep-materialization.R#L19), which is not currently run in CI. And it may have been a bad idea there.

But we have lots of cases where values are not `str`s.